### PR TITLE
Update Minimum Required Python Version to 3.9

### PR DESCRIPTION
**Pull request by Alrobbertz**
This PR updates the minimum required version of Python to 3.9 that is supported for the HERMES Core Package. It also updates the CI Pipelines to this version from older versions, and removes testing support for older versions of Python.

### CODE REVIEW

File: .readthedocs.yml [hunk 1]
```diff
@@ -22,10 +26,10 @@ formats:
...
-        - docs
-        - all
\ No newline at end of file
+# python:
+#   install:
+#     - method: pip
+#       path: .
+#       extra_requirements:
+#         - docs
+#         - all
```
Comment (dbarrous) [inline .readthedocs.yml:35]: Everything looks good, although small nit do we need this commented-out code if we will be handling it in the code block above?

**Review by dbarrous — APPROVED** [review]

File: .readthedocs.yml (hunk 1 above)
Comment (Alrobbertz) [reply .readthedocs.yml:35]: I thought about leaving it as a reminder to re-enable this approach once we have a more reliable way to install Spacepy. I should probably add this note as a comment in that area though so it doesn't just look like commented out code. 

We'll see what Steven thinks.

Comment (ehsteve) [reply .readthedocs.yml:35]: Fine with me.

**Review by ehsteve — APPROVED** [review]
Looks great! Thanks @Alrobbertz.

**Pull request merged**